feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,118 @@
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._base_transformers.base_numerical import BaseNumericalTransformer
+from feature_engine._base_transformers.mixins import (
+    FitFromDictMixin,
+    GetFeatureNamesOutMixin,
+)
+from feature_engine._check_input_parameters.check_init_input_params import (
+    _check_param_drop_original,
+)
+
+
+class DistanceFeatures(
+    BaseNumericalTransformer, FitFromDictMixin, GetFeatureNamesOutMixin
+):
+    EARTH_RADIUS: float = 6371.  # radius of Earth in kms
+
+    def __init__(
+            self,
+            a_latitude: str,
+            a_longitude: str,
+            b_latitude: str,
+            b_longitude: str,
+            output_column_name: Union[str, None] = None,
+            drop_original: bool = False,
+    ) -> None:
+
+        self.a_latitude = self._check_column_name(a_latitude)
+        self.a_longitude = self._check_column_name(a_longitude)
+        self.b_latitude = self._check_column_name(b_latitude)
+        self.b_longitude = self._check_column_name(b_longitude)
+
+        self.output_column_name = self._check_column_name(column_name=output_column_name)
+
+        _check_param_drop_original(drop_original=drop_original)
+        self.drop_original = drop_original
+
+        self.variables = None
+
+    @staticmethod
+    def _check_column_name(column_name: str) -> str:
```

Comment:
We allow integers as column names. I think I introduced this because it helps accommodate some of the compatibility tests with sklearn. In short, I don't think we need this check. We "outsource" this to pandas.

Could we remove?